qemu/tests/in_place_upgrade.py

```diff
@@ -90,8 +95,17 @@ def run(test, params, env):
         upgrade_test.post_upgrade_restore(test)
         # post checking
         upgrade_test.post_upgrade_check(test, post_release)
+        post_rhel_ver = upgrade_test.run_guest_cmd(check_rhel_ver)
         vm.verify_kernel_crash()
     finally:
-        if upgrade_test.session:
-            upgrade_test.session.close()
-        vm.destroy()
+        vm.graceful_shutdown(timeout=300)
+        try:
+            image_name = params.objects("images")[0]
+            image_params = params.object_params(image_name)
+            old_name = storage.get_image_filename(image_params,
+                                                  data_dir.get_data_dir())
```

Comment:
`data_dir.get_data_dir()` --> `params.get("images_base_dir", data_dir.get_data_dir())`

Reply:
Thanks, updated !

Reply:
Could you help to review it again ? Thanks !